torch/_higher_order_ops/cond.py

```diff
@@ -183,7 +226,16 @@ def trace_cond(proxy_mode, func_overload, pred, true_fn, false_fn, operands):
     for i in range(0, len(flat_true_outs)):
         true_out = flat_true_outs[i]
         false_out = flat_false_outs[i]
-        if true_out.meta["tensor_meta"] != false_out.meta["tensor_meta"]:
+        # TODO: If a torch nn module such as Linear or GRUCell is used, then the
+        # meta data of the output is None and cannot be compared
+        # TODO: If inside the dictionary, inside the list, the first element
+        # is composed of the multiplication, then the requires_grad attribute is
+        # set to False and thus the tracing of the cond errors out.
+        if (
+            true_out is not None
```

Comment:
If one branch return None but the other branch is not returning None, we should also error out. Can try to fix the error message to include this case.  We can have a expectedRaise test for this corner case.

Reply:
I added an additional [testcase](https://github.com/pytorch/pytorch/blob/630549a7e12c32b0f0f9a366c53e247b9a32a230/test/functorch/test_control_flow.py#L799) for this behavior